File: src/ProjectR/content/scripts/afflictions/Passives/RiseOfThePhoenix.lua
function GetName()
  return "Rise of the Phoenix"
end

function GetType()
  return Passive
end

function GetHookPoints()
  return { Dying }
end

function OnAttachment(char)
end

function OnRemoval(char)
end

function Trigger(hookPoint, char)
  if hookPoint ~= TurnTriggered then
    return
  end

  char:Heal(char:GetMaxHP())
  char:UseMP(-200)

  for i = HP, SIL do
    char:BuffStat(i, 2)
  end

  this:RemoveFrom(char)
end

File: src/ProjectR.Scripting/content/scripts/afflictions/Passives/RiseOfThePhoenix.cs
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Afflictions
{
    public class RiseOfThePhoenix : Affliction
    {
        public override string Name { get { return "Rise of the Phoenix"; } }

        public override AfflictionType Type { get { return AfflictionType.Passive; } }

        protected override HookPoint[] HookPoints
        {
            get
            {
                return new[]
                {
                    HookPoint.Dying, 
                };
            }
        }

        protected override void OnDying(ICharacter character)
        {
            character.Heal(character.MaxHP);
            character.UseMP(-200);

            for (var i = Stat.HP; i <= Stat.SIL; ++i)
            {
                character.BuffStat(i, 2);
            }

            RemoveFrom(character);
        }
    }
}